test/e2e/member_operator_test.go

```diff
@@ -0,0 +1,115 @@
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+	"github.com/codeready-toolchain/member-operator/pkg/common"
+	userv1 "github.com/openshift/api/user/v1"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestUserAccount(t *testing.T) {
+	userAccList := &toolchainv1alpha1.UserAccountList{}
+
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, userAccList)
+	require.NoError(t, err, "failed to add custom resource scheme to framework")
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err, "failed to initialize cluster resources")
+	t.Log("Initialized cluster resources")
+
+	namespace, err := ctx.GetNamespace()
+	require.NoError(t, err, "failed to get namespace where operator needs to run")
+
+	// get global framework variables
+	f := framework.Global
+	client := f.Client.Client
+
+	// wait for operator to be ready
+	err = e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "member-operator", 1, retryInterval, timeout)
+	require.NoError(t, err, "failed while waiting for operator deployment")
+
+	t.Log("member-operator is ready and running state")
+
+	// create useraccount
+	userAcc := newUserAcc(t, f, ctx)
+	err = f.Client.Create(context.TODO(), userAcc, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	require.NoError(t, err)
+	t.Logf("user account '%s' created", userAcc.Name)
+
+	t.Run("verify_useraccount_ok", func(t *testing.T) {
+		err := verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_user_ok", func(t *testing.T) {
+		user := &userv1.User{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name}, user)
+		require.NoError(t, err)
+
+		err = client.Delete(context.TODO(), user)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
+
+	t.Run("delete_identity_ok", func(t *testing.T) {
+		identity := &userv1.Identity{}
+		err := client.Get(context.TODO(), types.NamespacedName{Name: common.ToIdentityName(userAcc.Spec.UserID)}, identity)
+		require.NoError(t, err)
+
+		err = client.Delete(context.TODO(), identity)
+		require.NoError(t, err)
+
+		err = verifyResources(t, f, userAcc)
+		assert.NoError(t, err)
+	})
```

Comment:
Let's also add some tests for mapping "deletion". Basically it's just some editing on User or Identity which will break the mapping and see if it's being restored by the controller.

Reply:
The cascade deletion / garbage collection is handled by Kubernetes automatically if we set owner reference correctly. So it should already work.

Reply:
done [9b1b272](https://github.com/codeready-toolchain/member-operator/pull/26/commits/9b1b272ec26e4f6bfef703ca6a6d861e9818f2bb)